woocommerce/woocommerce-ios
# Jetpack CP: Error handling for install Jetpack

Part of #5365


### Description

This PR adds error handling for Jetpack install:
- Different title and message on Install Jetpack screen when installation fails
- New action buttons for when install error occurs: show WPAdmin and contact support. WPAdmin action is replaced with Retry Connection if plugin has been installed and activated successfully.
- Always shows Close button as a way to back out from the flow since it's impossible to dismiss the screen by swiping down on landscape mode.
- Checks Jetpack plugin status again after dismissing WPAdmin web view.

Things to improve in the next PRs:
- Remove hardcoded WPAdmin URL by updating Remote and Storage layers for Site to get the `admin_url`.
- Probably copies for the connection error - I made these up to complete the concept.

### Testing instructions

Prerequisite: the account is connected to at least a JCP site (setup tips in p1636091588189400-slack-C6H8C3G23) and a Jetpack site.
- Launch the app and log in to a JCP site if needed
- Open install Jetpack flow by tapping Jetpack benefit banner on My Store tab if it's visible, or follow Settings > Install Jetpack.
- Proceed to Get Started to get the installation started. 
- Notice that with good internet connection, you should be able to get through installation and activation steps. In connection step, you'll get error due to issue with `woocommerce_is_active` option mentioned in p91TBi-6EK-p2.
- Tapping on Contact Support action on error screen should show Zendesk view.
- Tapping on Retry Connection action should show connection step.
- To test error with installation step, repeat steps 1-2. Then turn off the wifi before proceeding to step 3.
- Notice that after the request times out, you'll see error screen with action button to install in WPAdmin. Turn on the wifi again and follow the page to install Jetpack.
- After activating Jetpack on the webview, dismiss it. You should then be navigated to connection step again.

### Demo

The video is a bit too long since I had to disconnect the wifi to force the request failure and then had trouble getting connection again 😅. Feel free to skip from 00:08 to 02:50. 
https://user-images.githubusercontent.com/5533851/144540310-76e26eaf-6f6e-40d8-90bb-0170dc51572f.mp4

---
